### PR TITLE
Add a new community pipeline

### PR DESCRIPTION
**Pull request by nagolinc**
This PR adds an img2img pipeline for Latent Consistency Models: https://github.com/luosiallen/latent-consistency-model/issues/8#issuecomment-1773970030

examples/community/latent_consistency_img2img.py

which can be called like this

import torch
from diffusers import DiffusionPipeline

pipe = DiffusionPipeline.from_pretrained(
                "SimianLuo/LCM_Dreamshaper_v7", custom_pipeline="latent_consistency_txt2img", custom_revision="main")

            # To save GPU memory, torch.float16 can be used, but it may compromise image quality.
pipe.to(torch_device="cuda", torch_dtype=torch.float32)

img2img=LatentConsistencyModelPipeline_img2img(
    vae=pipe.vae,
    text_encoder=pipe.text_encoder,
    tokenizer=pipe.tokenizer,
    unet=pipe.unet,
    #scheduler=pipe.scheduler,
    scheduler=None,
    safety_checker=None,
    feature_extractor=pipe.feature_extractor,
    requires_safety_checker=False,
)

img = Image.open("thisismyimage.png")

result = img2img(prompt,img,strength,num_inference_steps=4)

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/community/latent_consistency_img2img.py [hunk 1]
```diff
@@ -0,0 +1,819 @@
...
+from diffusers.utils.torch_utils import randn_tensor
+
+
+import PIL.Image
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class LatentConsistencyModelPipeline_img2img(DiffusionPipeline):
```
Comment (patrickvonplaten) [inline examples/community/latent_consistency_img2img.py:42]:
```suggestion
class LatentConsistencyModelImg2ImgPipeline(DiffusionPipeline):
```

File: examples/community/latent_consistency_img2img.py [hunk 2]
```diff
@@ -0,0 +1,819 @@
...
+    # Convert alphas_bar_sqrt to betas
+    alphas_bar = alphas_bar_sqrt**2  # Revert sqrt
+    alphas = alphas_bar[1:] / alphas_bar[:-1]  # Revert cumprod
+    alphas = torch.cat([alphas_bar[0:1], alphas])
+    betas = 1 - alphas
+
+    return betas
+
+
+class LCMScheduler_withTimestamp(SchedulerMixin, ConfigMixin):
```
Comment (patrickvonplaten) [inline examples/community/latent_consistency_img2img.py:480]:
```suggestion
class LCMSchedulerWithTimestamp(SchedulerMixin, ConfigMixin):
```

File: examples/community/latent_consistency_img2img.py [hunk 3]
```diff
@@ -0,0 +1,819 @@
...
+        safety_checker: StableDiffusionSafetyChecker,
+        feature_extractor: CLIPImageProcessor,
+        requires_safety_checker: bool = True,
+    ):
+        super().__init__()
+
+        scheduler = (
+            scheduler
+            if scheduler is not None
+            else LCMScheduler_withTimestamp(
```
Comment (patrickvonplaten) [inline examples/community/latent_consistency_img2img.py:61]:
```suggestion
            else LCMSchedulerWithTimestamp(
```

File: examples/community/latent_consistency_img2img.py [hunk 4]
```diff
@@ -0,0 +1,819 @@
...
+class LatentConsistencyModelPipeline_img2img(DiffusionPipeline):
+    _optional_components = ["scheduler"]
+
+    def __init__(
+        self,
+        vae: AutoencoderKL,
+        text_encoder: CLIPTextModel,
+        tokenizer: CLIPTokenizer,
+        unet: UNet2DConditionModel,
+        scheduler: "LCMScheduler_withTimestamp",
```
Comment (patrickvonplaten) [inline examples/community/latent_consistency_img2img.py:51]:
```suggestion
        scheduler: "LCMSchedulerWithTimestamp",
```

**Review by patrickvonplaten — COMMENTED** [review]
Cool addition! Could we maybe also add a small entry to the README here: https://github.com/huggingface/diffusers/tree/main/examples/community#latent-consistency-pipeline

**Comment by nagolinc** [comment]
>  would reformat /home/runner/work/diffusers/diffusers/examples/community/latent_consistency_img2img.py

which formatter do I need to run?

**Comment by patrickvonplaten** [comment]
`make style` => I can run it on `"main"`

**Comment by nagolinc** [comment]
thanks!

**Comment by Zaniyar** [comment]
> Cool addition! Could we maybe also add a small entry to the README here: https://github.com/huggingface/diffusers/tree/main/examples/community#latent-consistency-pipeline

```
 from diffusers import Diffusionpipeline
ImportError: cannot import name 'Diffusionpipeline' from 'diffusers' (/home/user/anaconda3/lib/python3.11/site-packages/diffusers/__init__.py)
```

I tried: 
$ conda install -c conda-forge diffusers
$ conda install diffusers
$ pip install diffusers
$ pip3 install diffusers

it is installed, but can't be imported in python - any ideas?

**Pull request merged**